syntax = "proto3";

message ClientCommand {
    int64 command_id = 1;
    int64 user_id = 2;
    oneof command {
        WaitlistAdd waitlist_add = 3;
        WaitlistMove waitlist_move = 4;
        WaitlistDelete waitlist_delete = 5;
        Play play = 6;
        Pause pause = 7;
        Pause jump = 8;
        AdjustVolume adjust_volume = 9;
        GetWaitlist get_waitlist = 10;
        GetMLibrary get_mlibrary = 11;
        GetActiveSong get_active_song = 12;
    }
}

message WaitlistAdd {
    int64 sid = 1;
}

message WaitlistMove {
    int64 wid = 1;
    int64 offset = 2;
}

message WaitlistDelete {
    int64 wid = 1;
}

message Play {
    int64 sid = 1;
}

message Pause {

}

message Jump {
    int64 time = 1;
}

message AdjustVolume {
    int64 volume = 1;
}

message GetWaitlist {

}

message GetMLibrary {
    int64 page = 1;
}

message GetActiveSong {

}